bridges/snowbridge/primitives/core/src/lib.rs

```diff
@@ -163,4 +163,30 @@ impl DescribeLocation for DescribeHere {
 
 /// Creates an AgentId from a Location. An AgentId is a unique mapping to a Agent contract on
 /// Ethereum which acts as the sovereign account for the Location.
-pub type AgentIdOf = HashedDescription<H256, (DescribeHere, DescribeFamily<DescribeAllTerminal>)>;
+pub type AgentIdOf =
+	HashedDescription<AgentId, (DescribeHere, DescribeFamily<DescribeAllTerminal>)>;
+
+#[derive(Clone, Default, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+pub struct AssetRegistrarMetadata {
+	pub name: Vec<u8>,
+	pub symbol: Vec<u8>,
```

Comment:
We should a constraint that `name` and `symbol` should not be longer than 32 bytes. To prevent attacks that could block a channel by consuming an unbounded amount of gas when the metadata is stored on the ethereum side.

```suggestion
	pub name: BoundedVec<u8, 32>,
	pub symbol: BoundedVec<u8, 32>,
```

Reply:
Cool. Though please check that the `GasMeter` accurately returns the worst case gas for instantiating an ERC20 with a name and symbol limited to 32 bytes each.

Reply:
Smoke test with name and symbol limited to 32 bytes in https://github.com/Snowfork/snowbridge/pull/1155/commits/001c990428bf47ee74ae7c535a22735df3843513 which shows that **Gas used: 957953** for register the PNA. 

Less than https://github.com/Snowfork/polkadot-sdk/blob/79e62fa1e8b326fcec6c8ae40973ed1de94c92b9/bridges/snowbridge/primitives/core/src/outbound.rs#L461 the maximum_dispatch_gas: 1_200_000 so should be safe.